common/types.go

```diff
@@ -158,6 +176,142 @@ func (h UnprefixedHash) MarshalText() ([]byte, error) {
 	return []byte(hex.EncodeToString(h[:])), nil
 }
 
+/////////// ExtHash
+
+type (
+	// ExtHash is an extended hash composed of a 32 byte Hash and a 6 byte Nonce
+	// ExtHash is used as the identifier of Merkle Patricia Trie nodes under State Pruning scheme.
+	// The Hash component shall be equal with and without State Pruning to make sure
+	// merkle proofs are unchanged. The Nonce component shall be a fresh number whenever a new
+	// node is created. See KIP-112 for details.
+	ExtHash      [ExtHashLength]byte
+	ExtHashNonce [ExtHashNonceLength]byte
+)
+
+// BytesToExtHash converts the byte array b to ExtHash.
+// If len(b) is 0 to 32, then interpret b as Hash and then attaches LegacyNonce.
+// Otherwise b is interpreted as ExtHash. If len(b) is larger than len(ExtHash), b
+// will be cropped from the left.
+func BytesToExtHash(b []byte) ExtHash {
```

Comment:
Is it okay this function doesn't check the length strictly? 
This function allows an input byte array having less than 32 lengths, then someone may use the leading zero removed byte array for hash. What if the array has 26 bytes for hash and  6 bytes for an extHashNonce? It will consider w

Reply:
Modified to strictly check if the length is 0, HashLength, or ExtHashLength.